src/Infrastructure/Vine/VineTaxonFilters.php

```
@@ -0,0 +1,187 @@
+    private TaxonomyRepository $taxonomyRepository;
+
+    public function __construct(TraderConfig $traderConfig, TaxonTreeRepository $taxonTreeRepository, TaxonomyRepository $taxonomyRepository)
+    {
+        $this->taxonTreeRepository = $taxonTreeRepository;
+        $this->traderConfig = $traderConfig;
+        $this->taxonomyRepository = $taxonomyRepository;
+    }
+
+    public function getAvailableFilters(Locale $locale, array $scopedTaxonIds): array
+    {
+        $mainCategoryTaxonomyId = $this->traderConfig->getMainCategoryTaxonomyId();
```

PHPMD (2.15.0)

Avoid unused local variables such as '$mainCategoryTaxonomyId'. (undefined)

(UnusedLocalVariable)